addons/test_event_full/static/src/js/tests/test_template_reference_field_widget.js

```diff
@@ -0,0 +1,164 @@
+                            id: 3,
+                            template_ref: "some.template,1",
+                        },
+                    ],
+                },
+                "mail.template": {
+                    fields: {
+                        name: {
+                            string: "Name",
+                            type: "text",
+                        },
+                    },
+                    records: [{ id: 1, name: "Mail Template 1" }],
+                },
+                "sms.template": {
+                    fields: {
+                        name: {
+                            string: "Name",
+                            type: "text",
+                        },
+                    },
+                    records: [{ id: 1, name: "SMS template 1" }],
+                },
+                "some.template": {
+                    fields: {
+                        name: {
+                            string: "Name",
+                            type: "text",
+                        },
+                    },
+                    records: [{ id: 1, name: "Some Template 1" }],
+                },
+                "ir.model": {
+                    fields: {
+                        model: { string: "Model", type: "char" },
+                    },
+                    records: [
+                        {
+                            id: 17,
+                            name: "Partner",
+                            model: "partner",
+                        },
+                        {
+                            id: 20,
+                            name: "Product",
+                            model: "product",
+                        },
+                        {
+                            id: 21,
+                            name: "Partner Type",
+                            model: "partner_type",
+                        },
+                    ],
+                    onchanges: {},
+                },
+            },
+        };
+
+        setupViewRegistries();
+
+        patchWithCleanup(browser, {
+            setTimeout: (fn) => fn(),
+        });
+    });
+
+    QUnit.test("Reference field displays right icons", async function (assert) {
+        assert.expect(12);
+        await makeView({
+            type: "list",
+            resModel: "mail",
+            serverData,
+            arch: `
+            <tree editable="top">
+                <field name="template_ref" widget="EventMailTemplateReferenceField"/>
+            </tree>`,
+        });
+
+        assert.containsN(target, ".o_field_cell", 3);
+        assert.containsN(target, ".o_field_cell.o_EventMailTemplateReferenceField_cell", 3);
+        let cells = [...target.querySelectorAll(".o_field_cell")];
+        assert.containsOnce(cells[0], ".fa-envelope");
+        assert.containsOnce(cells[1], ".fa-mobile");
+        assert.containsNone(cells[2], ".fa-envelope");
+        assert.containsNone(cells[2], ".fa-mobile");
```

Comment:
So in this context each "cell" matches a field of the list?
And as we only have one field, it's the cell of our new referencefield, for each of the 3 records?